framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -487,6 +488,11 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
                                     + " should be one of: -Ainfer=jaifs, -Ainfer=stubs");
             }
             wholeProgramInference = new WholeProgramInferenceScenes(this);
+            if (!checker.hasOption("warns")) {
+                // Without -Awarns, the inference output may be incomplete, because javac halts
+                // after issuing an error.
+                checker.message(Diagnostic.Kind.ERROR, "Do not supply -Ainfer without -Awarns");
```

Comment:
What about:

```suggestion
                checker.message(Diagnostic.Kind.ERROR, "Do not supply -Ainfer without -Awarns. Without -Awarns, javac will halt after issuing errors, which may lead to incomplete inference results.");
```

Reply:
It's a good inclination to provide information to the user.  I agree in general.  Thanks for the suggestion.

Two countervailing principles are:
 * keep messages short, which makes them easier to read and interpret; provide only the essential information.
 * avoid duplicating text, for example in the manual and in the warning message, unless necessary.

I think the rationale is easy to find in the manual if the user wants it, so I'm going to leave the message short for now.  If users get confused, I will admit I was wrong and we can make the message more lengthy.